avocado/utils/data_structures.py

```diff
@@ -33,6 +34,24 @@ def ordered_list_unique(object_list):
     return [x for x in object_list if not (x in seen or seen_add(x))]
 
 
+def geometric_mean(values):
+    """
+    Evaluates the geometric mean for a list of numeric values.
+
+    :param values: List with values.
+    :return: Single value representing the geometric mean for the list values.
+    :see: http://en.wikipedia.org/wiki/Geometric_mean
+    """
+    try:
+        values = [int(value) for value in values]
+    except ValueError:
+        return None
```

Comment:
Hmm, wouldn't it be better to raise the `ValueError` in case of incorrect input? That's what I'd except, actually.

In tests, that'd result in traceback explaining what went wrong and exactly where and if one expects those failures, he can still use `try/except` to allow empty results.